### DISCUSSION TITLE
Union + positional_order_by + limit

### DISCUSSION THREAD

**Discussion by nemasu** [diesel-rs/diesel#4336] [Q&A] [answered]
Hello!

I'm trying to use a union query with positional_order_by and limit.
It compiles fine until I add limit(), I looked at the examples, and I think it should work?

The failing line is here
https://github.com/nemasu/puddle-farm/blob/a3f12a521f0337a3c7d562a206d8aa6ab166c0f4/src/main.rs#L364
(ignore the other 'order by' and 'limit', they would not be used)

The error is quite long, but it starts with:
`the method 'limit' exists for struct 'PositionalOrderClause<CombinationClause<Union, Distinct, SelectStatement<..., ..., ..., ...>, ...>, ...>', but its trait bounds were not satisfied`.

and ends with
`positional_order_dsl.rs(24, 1): doesn't satisfy '_: QueryDsl' or _: Table'`

Thanks!

**Comment by weiznich** [comment, marked as answer]
Support for `limit()` for union queries landed only recently on the master branch (https://github.com/diesel-rs/diesel/pull/4245), so it's not available on any released diesel version yet. That means this behavior is currently expected.

**Reply by nemasu** [reply]
OK that makes sense. Thank you.

**Reply by dessalines** [reply]
Was looking for this also, thx!
